code/game/gamemodes/objective.dm

```diff
@@ -159,11 +159,39 @@ GLOBAL_LIST_EMPTY(objectives)
 	var/datum/mind/receiver = pick(get_owners())
 	if(receiver?.current)
 		if(ishuman(receiver.current))
-			var/mob/living/carbon/human/H = receiver.current
+			var/mob/living/carbon/human/receiver_current = receiver.current
 			var/list/slots = list("backpack" = ITEM_SLOT_BACKPACK)
 			for(var/eq_path in special_equipment)
-				var/obj/O = new eq_path
-				H.equip_in_one_of_slots(O, slots)
+				var/obj/object = new eq_path
```

Comment:
1. Use `as anything` in the loop rather than reassigning it here.
2. This is a useless variable name, what object is it?

Reply:
`object` is a concrete instance of `eq_path`. If I were to simply say `eq_path = new eq_path`, then `equip_in_one_of_slots` will qdel `eq_path`, resulting in a `null` getting added to `failed_special_equipment`. For clarity's sake, I will however rename `eq_path` to `equipment_path` and `object` to `equipment_object`

Reply:
I didn't notice the `new`, you're right.